week05/errors_test.go

```diff
@@ -0,0 +1,176 @@
+package demo
+
+import (
+	"reflect"
+	"testing"
+)
+
+func TestErrTableNotFoundError(t *testing.T) {
+	t.Parallel()
+
+	tn := "testtable"
+	testError := ErrTableNotFound{
+		table:		tn,
+	}
+
+	exp := "table not found " + tn
+	act := testError.Error()
+
+	if exp != act {
+		t.Fatalf("expected %s, got %s", exp, act)
+	}
+}
+
+func TestErrTableNotFound(t *testing.T) {
+	t.Parallel()
+
+	tn := "testtable"
+	testError := ErrTableNotFound{
+		table:		tn,
+	}
+
+	exp := tn
+	act := testError.TableNotFound()
+
+	if exp != act {
+		t.Fatalf("expected %s, got %s", exp, act)
+	}
+}
+
+func TestIsErrTableNotFound(t *testing.T) {
+	t.Parallel()
+	tn := "testtable"
+
+	testError := ErrTableNotFound{
+		table:		tn,
+	}
+
+	if !IsErrTableNotFound(testError) {
+		t.Fatalf("error asserting type")
+	}
+}
+
+
+func TestErrNoRowsError(t *testing.T) {
+	t.Parallel()
+
+	tn := "testtable"
+	testError := errNoRows{
+		clauses:	Clauses{"id": 1, "name": "John"},
+		table:		tn,
+	}
+
+	exp := "[" + tn + "] no rows found\nquery: " + testError.clauses.String()
+	act := testError.Error()
+
+	if exp != act {
+		t.Fatalf("expected %s, got %s", exp, act)
+	}
+}
+
+func TestErrNoRowsClauses(t *testing.T) {
+	t.Parallel()
+
+	tn := "testtable"
+	testClause 	:= Clauses{"id": 2, "name": "Jane"} 
+
+	table := []struct {
+		name 	string
+		err  	errNoRows
+	}{
+		{name: "empty clauses", err: errNoRows{testClause, tn}},
+		{name: "existing clauses", err: errNoRows{nil, tn}},
+	}
+
+	for _, tt := range table {
+		t.Run(tt.name, func(t *testing.T) {
+
+			act := tt.err.Clauses()
+			var exp Clauses
+			
+			if len(tt.err.clauses) == 0 {
+				exp = Clauses{}
+			} else {
+				exp = testClause
+			}
```

Comment:
In Go, we try to remove `else` statements when we can, often they can be removed without little effort. Here, you are probably missing an `exp Clauses` field on your table struct.